src/ImageSharp/Processing/DefaultImageProcessorContext{TPixel}.cs

```diff
@@ -67,16 +67,23 @@ public IImageProcessingContext ApplyProcessor(IImageProcessor processor, Rectang
                 // This will only work if the first processor applied is the cloning one thus
                 // realistically for this optimization to work the resize must the first processor
                 // applied any only up processors will take the double data path.
-                if (processor.CreatePixelSpecificProcessor(this.source, rectangle) is ICloningImageProcessor<TPixel> cloningImageProcessor)
+                using (IImageProcessor<TPixel> specificProcessor = processor.CreatePixelSpecificProcessor(this.source, rectangle))
                 {
-                    this.destination = cloningImageProcessor.CloneAndApply();
-                    return this;
+                    if (specificProcessor is ICloningImageProcessor<TPixel> cloningImageProcessor)
+                    {
+                        this.destination = cloningImageProcessor.CloneAndApply();
+                        return this;
+                    }
```

Comment:
If the processor is not a `ICloningImageProcessor<TPixel>` we are disposing and re-creating it.

Reply:
Good catch, will fix. 👍

Reply:
Maybe I'm just too tired but I cannot see a workaround here. The processors are created from different images.

Reply:
Indeed, haven't noticed that!
Makes it harder, I will also keep  thinking tomorrow.